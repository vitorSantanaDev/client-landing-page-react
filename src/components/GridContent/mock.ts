export default {
  title: 'Title text example',
  content: `<p>Lorem ipsum dolor sit amet, <a href="#">consectetur adipiscing</a> elit. Sed ac lacus pretium urna malesuada dignissim. Sed risus tortor, sagittis eget sem ut, porta imperdiet magna. In fringilla nisi enim. Aenean mollis vehicula nulla, non vulputate eros pretium eget. Vivamus tincidunt accumsan massa non convallis. Aliquam molestie efficitur faucibus. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; Vestibulum suscipit convallis mi, vitae ornare dolor hendrerit quis. Nam gravida orci posuere nisi imperdiet posuere. Phasellus quis blandit turpis, eu feugiat velit. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Integer ac malesuada quam. Aliquam eu finibus massa.</p>

  <p>Donec pulvinar volutpat ante quis porttitor. In hac habitasse platea dictumst. Sed eu placerat turpis, vitae pulvinar odio. Pellentesque viverra posuere enim, eu dapibus quam congue sit amet. Sed iaculis eleifend scelerisque. Curabitur ut elit at sem cursus euismod. Praesent ornare, urna vitae sollicitudin tincidunt, sem odio accumsan velit, quis iaculis neque elit vel ante. Suspendisse rhoncus felis at arcu blandit facilisis. Cras euismod varius nibh, a cursus nisi. Integer laoreet posuere sem vel pulvinar. Etiam vel blandit libero. Praesent consectetur tellus et justo lobortis rhoncus.</p>
  `,
  background: false
}
