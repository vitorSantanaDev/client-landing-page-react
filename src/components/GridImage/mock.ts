export default {
  background: false,
  title: 'My grid',
  description:
    'Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde',
  grid: [
    {
      image: {
        altText: 'Teste 1',
        srcImage: 'https://source.unsplash.com/random/800x800?r=1'
      }
    },
    {
      image: {
        altText: 'Teste 2',
        srcImage: 'https://source.unsplash.com/random/800x800?r=2'
      }
    },
    {
      image: {
        altText: 'Teste 3',
        srcImage: 'https://source.unsplash.com/random/800x800?r=3'
      }
    },
    {
      image: {
        altText: 'Teste 4',
        srcImage: 'https://source.unsplash.com/random/800x800?r=4'
      }
    },
    {
      image: {
        altText: 'Teste 5',
        srcImage: 'https://source.unsplash.com/random/800x800?r=5'
      }
    },
    {
      image: {
        altText: 'Teste 6',
        srcImage: 'https://source.unsplash.com/random/800x800?r=6'
      }
    }
  ]
}
