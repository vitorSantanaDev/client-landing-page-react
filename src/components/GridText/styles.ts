import styled, { css } from 'styled-components'
import { Wrapper as Heading } from '../Heading/styles'
import { Wrapper as TextComponent } from '../TextComponent/styles'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`
export const Grid = styled.div`
  ${({ theme }) => css`
    width: 100%;
    overflow: hidden;
    display: grid;
    counter-reset: grid-counter;
    gap: ${theme.spacings.large};
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  `}
`
export const GridElement = styled.div`
  ${() => css`
    ${Heading} {
      left: 5rem;
      position: relative;
    }

    ${Heading}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);

      top: -2.3rem;
      left: -5rem;
      position: absolute;
      transform: rotate(10deg);

      font-size: 7rem;
    }
  `}
`
