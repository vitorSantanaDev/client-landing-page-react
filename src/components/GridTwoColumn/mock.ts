export default {
  title: 'Grid two column',
  text: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Ad vero, accusantium eum exercitationem in ex quo veniam sed totam, quos ipsam at. Facilis a labore alias sed, natus dolor temporibus?',
  srcImage: 'assets/images/javascript.svg'
}
