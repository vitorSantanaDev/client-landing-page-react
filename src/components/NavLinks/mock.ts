export default [
  { children: 'First Link', link: '#target_1', newTab: false },
  { children: 'Second Link', link: '#target_2', newTab: true },
  { children: 'Tertiary Link', link: '#target_3', newTab: false },
  { children: 'Last Link', link: '#target_4', newTab: true },
  { children: 'Test Link', link: '#target_5', newTab: false },
  { children: 'Contact Link', link: '#target_6', newTab: false },
  { children: 'About Link', link: '#target_7', newTab: true }
]
