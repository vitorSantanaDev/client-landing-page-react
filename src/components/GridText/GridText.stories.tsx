import React from 'react'
import { ComponentStory, ComponentMeta } from '@storybook/react'
import GridText from '.'
import mock from './mock'

export default {
  title: 'GridText',
  component: GridText,
  args: { ...mock }
} as ComponentMeta<typeof GridText>

export const Default: ComponentStory<typeof GridText> = (args) => (
  <GridText {...args} />
)
