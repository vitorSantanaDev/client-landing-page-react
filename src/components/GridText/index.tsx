import React from 'react'

import Heading from '../Heading'
import SectionBackground from '../SectionBackground'
import TextComponent from '../TextComponent'
import * as S from './styles'

interface IGridText {
  title: string
  background?: boolean
  description: string
  sectionId?: string
  grid: {
    title: string
    description: string
  }[]
}

const GridText: React.FC<IGridText> = ({
  title,
  background,
  description,
  grid,
  sectionId
}) => (
  <SectionBackground sectionId={sectionId} background={background}>
    <S.Wrapper>
      <Heading size="huge" uppercase colorDark={!background} as="h2">
        {title}
      </Heading>
      <TextComponent>{description}</TextComponent>
      <S.Grid>
        {grid.map((element) => (
          <S.GridElement key={element.title}>
            <Heading size="medium" colorDark={!background} as="h3">
              {element.title}
            </Heading>
            <TextComponent>{element.description}</TextComponent>
          </S.GridElement>
        ))}
      </S.Grid>
    </S.Wrapper>
  </SectionBackground>
)

export default GridText
