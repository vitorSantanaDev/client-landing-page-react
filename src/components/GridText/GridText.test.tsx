import React from 'react'
import { screen } from '@testing-library/react'
import { renderTheme } from '../../styles/theme/render-theme'

import GridText from '.'
import mock from './mock'

describe('<GridText />', () => {
  it('should render the component GridText', () => {
    const { container } = renderTheme(<GridText {...mock} />)
    const headings = screen.getAllByRole('heading', { name: /teste/i })

    expect(headings).toHaveLength(3)
    expect(container.firstChild).toMatchSnapshot()
  })
  it('should not render the headings when grid prop is empyt', () => {
    const { container } = renderTheme(
      <GridText title="My grid" description="test" grid={[]} />
    )
    const headings = screen.queryAllByRole('heading', { name: /teste/i })

    expect(headings).toHaveLength(0)
    expect(container.firstChild).toMatchSnapshot()
  })
})
