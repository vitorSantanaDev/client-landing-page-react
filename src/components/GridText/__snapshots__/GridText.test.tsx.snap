// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridText /> should render the component GridText 1`] = `
.c4 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c9 {
  color: #0A1128;
  font-size: 3.2rem;
  text-transform: none;
}

.c1 {
  max-width: 1200px;
  margin: 0 auto;
  padding: 3.2rem;
}

.c0 {
  background-color: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  font-size: 2.4rem;
}

.c2 .c5 {
  margin-bottom: 6.4rem;
}

.c7 {
  counter-reset: grid-counter;
  display: grid;
  gap: 3.2rem;
  grid-template-columns: repeat(auto-fill,minmax(280px,1fr));
}

.c8 .c3 {
  left: 5rem;
  position: relative;
}

.c8 .c3::before {
  counter-increment: grid-counter;
  content: counter(grid-counter);
  top: -2.3rem;
  left: -5rem;
  position: absolute;
  -webkit-transform: rotate(10deg);
  -ms-transform: rotate(10deg);
  transform: rotate(10deg);
  font-size: 7rem;
}

@media (max-width:768px) {
  .c4 {
    font-size: 4.0rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <h2
        class="c3 c4"
      >
        My grid
      </h2>
      <p
        class="c5 c6"
      >
        Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde
      </p>
      <div
        class="c7"
      >
        <div
          class="c8"
        >
          <h3
            class="c3 c9"
          >
            Teste 1
          </h3>
          <p
            class="c5 c6"
          >
            Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
          </p>
        </div>
        <div
          class="c8"
        >
          <h3
            class="c3 c9"
          >
            Teste 2
          </h3>
          <p
            class="c5 c6"
          >
            Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
          </p>
        </div>
        <div
          class="c8"
        >
          <h3
            class="c3 c9"
          >
            Teste 3
          </h3>
          <p
            class="c5 c6"
          >
            Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
